### ISSUE TITLE
grpc_tools_node_protoc_plugin plugin not found 

### ISSUE THREAD

**Issue by sri-vathsa** [agreatfool/grpc_tools_node_protoc_ts#100]
I am using an ubuntu 20 ARM machine and to install `grpc_tools_node_protoc_ts` globally.  Following is the output for the command 
```
/usr/bin/protoc-gen-ts -> /usr/lib/node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts
+ grpc_tools_node_protoc_ts@5.2.1
updated 1 package in 0.436s
```
But I could not use the `grpc_tools_node_protoc_plugin`(I used the `which grpc_tools_node_protoc_plugin` to check this).
Can someone help me out on how to get the plugin working on my machine?

**Comment by agreatfool** [comment]
try to replace the `which` statement to `./node_modules/.bin/grpc_tools_node_protoc_plugin`

**Comment by sri-vathsa** [comment]
There is no .bin folder in the node_modules folder and I tried searching for the file globally using ` sudo find /  -name "grpc_tools_node_protoc_plugin"` and the only result is `/usr/bin/grpc_tools_node_protoc_plugin` . When I run 
`file  /usr/bin/grpc_tools_node_protoc_plugin` I got the following output
```
/usr/bin/grpc_tools_node_protoc_plugin: broken symbolic link to ../lib/node_modules/grpc-tools/bin/protoc_plugin.js
```
Then I changed into the above `node_modules` folder to search for the `protoc_plugin.js` file using ` find .  -name "protoc_plugin.js"` and found no results.

**Comment by agreatfool** [comment]
`yarn add grpc-tools` to install it to your local repo. then try with `node_modules/.bin/...`

**Comment by agreatfool** [comment]
and you could submit an issue here: https://github.com/grpc/grpc-node

**Comment by sri-vathsa** [comment]
node grpc does not provide binaries for ARM according to [this](https://github.com/grpc/grpc-node/issues/1405). Is there any way I can use the plugin without installing the grpc-tools?

**Comment by agreatfool** [comment]
No... as you are trying to generate stub js codes from proto file, this action requires `grpc-tools`

**Comment by agreatfool** [comment]
I read through the ticket you pasted, that's not good, im switching to m1 chip mac soon.

**Comment by sri-vathsa** [comment]
Maybe this [ticket](https://github.com/grpc/grpc-node/issues/1737#issuecomment-814257861) would also help.

**Comment by Clement-Jean** [comment]
For those who are on M1 mac, i made the install work with `npm_config_target_arch=x64 npm i grpc-tools`. It is probably not the best solution but it is working fine.

**Comment by agavitalis** [comment]
> For those who are on M1 mac, i made the install work with `npm_config_target_arch=x64 npm i grpc-tools`. It is probably not the best solution but it is working fine.

This works
